src/Microsoft.AspNet.Mvc.Core/Routing/IUrlHelperFactory.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Mvc.Routing
```

Comment:
It was always intended to put `IUrlHelper` in .`Routing`. Starting a side-discussion about whether or not to move that.